chatlogger.lua

```diff
@@ -109,15 +131,62 @@ ExecuteCommand = {
 	topic = ( function()
 		local sErased, sUpdated = ( "<%s> Hub topic was erased by [ %%s ]." ):format( tConfig.sBotName ), ( "<%s> Hub topic was updated by [ %%s ] to %%s." ):format( tConfig.sBotName )
 		return function( tUser, sData, bIsPM )
-			if not ProfMan.GetProfilePermission( tUser.iProfile, 7 ) then return false end
-			if sData:len() == 0 then
+			if not CheckPermission( tUser.iProfile ) then return false end
+			if sData:len() == 0 or sData:lower() == "off" then
 				Core.SendToAll( sErased:format(tUser.sNick) )
-				SetMan.SetString( 10, "" )
-				return true
+				return ExecuteCommand.ticker( tUser, sData, bIsPM )
+			end
+			if tConfig.iTickerID then
+				TmrMan.RemoveTimer( tConfig.iTickerID )
+				tConfig.iTickerID = false
 			end
 			SetMan.SetString( 10, sData )
 			Core.SendToAll( sUpdated:format(tUser.sNick, sData) )
 			return true
 		end
 	end )(),
+
+	ticker = ( function()
+		local sReply = ( "<%s> Topics ticker has been activated." ):format( tConfig.sBotName )
+		return function( tUser, sData, bIsPM )
+			if not CheckPermission( tUser.iProfile ) then return false end
+			if tConfig.iTickerID then return false end
+			tConfig.iTickerID = TmrMan.AddTimer( tConfig.iTickerDelay )
+			return Reply( tUser, sReply, bIsPM )
+		end
+	end )(),
+
+	tickeradd = ( function()
+		local sFilePath, sReply, sTemplate = tConfig.sPath..tConfig.sFileName, ( "<%s> The topic has been added to tickers list." ):format( tConfig.sBotName ), "%s %s\n"
+		return function( tUser, sData, bIsPM )
+			if not CheckPermission( tUser.iProfile ) then return false end
+			local fTickerHandle, sNick, sTopic = io.open( sFilePath, "a+" ), sData:match "%-u (%S+) (.+)"
```

Comment:
The file is being opened in append mode. It gets created if not already present.

As for `-u` flag; it'd be too much trouble. Since barely 3-5 users have the permission to change topics; they can manage using `-u` in the beginning.